src/service-override/files.ts

```diff
@@ -194,8 +194,9 @@ class RegisteredFileSystemProvider extends Disposable implements IFileSystemProv
 
   public async readdir (resource: URI): Promise<[string, FileType][]> {
     const includedPaths = Array.from(this.files.keys())
-      .map(uri => extUri.relativePath(resource, URI.parse(uri))!)
-      .filter(path => !path.startsWith('..'))
+      .map(uri => URI.parse(uri))
+      .filter(uri => uri.path.startsWith(resource.path))
+      .map(uri => extUri.relativePath(resource, uri)!)
```

Comment:
For some reasons, `relativePath` returns the absolute path if the second parameter is not a child of the first parameter, and not a path starting with `..`

Reply:
Strange indeed. How did you find that problem in the first place?

Reply:
Found it while investigating the issue you've reported!

Reply:
Just curious. How did you identify this? Debugging in Chrome or what did you do? I had no starting point for analysing it.

Reply:
It was surprisingly easy! I just had to "pause script execution" in chrome dev tools to realize it was calling this method in loop recursively until a stack overflow occurs and then do it again